lib/jekyll/drops/drop.rb

```diff
@@ -85,12 +85,8 @@ def []=(key, val)
       # Returns an Array of strings which represent method-specific keys.
       def content_methods
         @content_methods ||= (
-          self.class.instance_methods \
-            - Jekyll::Drops::Drop.instance_methods \
-            - NON_CONTENT_METHODS
-        ).map(&:to_s).reject do |method|
-          method.end_with?("=")
-        end
+          self.class.instance_methods - Jekyll::Drops::Drop.instance_methods - NON_CONTENT_METHODS
+        ).map!(&:to_s).tap { |result| result.reject! { |method| method.end_with?("=") } }
```

Comment:
```suggestion
        ).map!(&:to_s).reject! { |method| method.end_with?("=") }
```

What's the reason to use `#tap` here? I think we can remove it now.

Reply:
Since [`Array#reject!`](https://ruby-doc.org/core-2.4.0/Array.html#method-i-reject-21) can return `nil` if the input array was changed, calling `#tap` ensures that the input array is returned.

Reply:
Gotcha! Thanks for the explanation – the bang (!) methods are oddities. 😄